locations/spiders/denmans_gb.py

```diff
@@ -0,0 +1,53 @@
+from scrapy import Spider
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+
+
+class DenmansGBSpider(Spider):
+    name = "denmans_gb"
+    item_attributes = {"brand": "Denmans", "brand_wikidata": "Q116508855"}
+    # This seems to return all stores regardless of lat long as long as it's in the UK?
+    start_urls = ["https://www.denmans.co.uk/den/store-finder/findNearbyStores?latitude=51&longitude=-0"]
+
+    def parse(self, response):
+        for store in response.json()["results"]:
+            store["address"]["street_address"] = ", ".join(
+                filter(
+                    None,
+                    [
+                        store["address"].pop("line1"),
+                        store["address"].pop("line2"),
+                        store["address"].pop("line3"),
+                    ],
+                )
+            )
+            store["address"]["country"] = store["address"]["country"]["isocode"]
+            item = DictParser.parse(store)
+            item["name"] = store["displayName"]
+            item["ref"] = store["name"]
+            # We can either pass in store, and it picks up the lat/long, or store["address"] and it gets the email and phone
+            item["email"] = store["address"]["email"]
```

Comment:
This one could be fixed by adding email1 to the DictParser, as they duplicate it in another field, or both could try obj, then fall back to obj["address"] if no results...

Reply:
Contact details in an address object does seem to be relatively common. So it'd be nice to support it, but I think the most common situation is it being in the root (`obj`). I'm nervous about this one because I don't think anyone looks at the stats of contact details so it we lost a bunch nobody would notice.

Adding `email1` seems like a clear win though.

Reply:
Understood, I'd think things like none/str checks would catch this, I've left it with just email1 for now anyway.